hbase-balancer/src/main/java/org/apache/hadoop/hbase/master/balancer/StochasticLoadBalancer.java

```diff
@@ -343,15 +343,15 @@ boolean needsBalance(TableName tableName, BalancerClusterState cluster) {
       sendRejectionReasonToRingBuffer(() -> getBalanceReason(calculatedTotal, calculatedMultiplier),
         costFunctions);
     }
-    if (LOG.isDebugEnabled()) {
-      LOG.debug("{} {}; total cost={}, sum multiplier={}; cost/multiplier to need a balance is {}",
-          balanced ? "Skipping load balancing because balanced" : "We need to load balance",
-          isByTable ? String.format("table (%s)", tableName) : "cluster",
-          total, sumMultiplier, minCostNeedBalance);
-      if (LOG.isTraceEnabled()) {
-        LOG.trace("Balance decision detailed function costs={}", functionCost());
-      }
+    LOG.info("{} {}; total cost={}, sum multiplier={}; cost/multiplier to need a balance is {}",
+      balanced ? "Skipping load balancing because balanced" :
+        "Start calculating moving plan without logging info for up to {} secs",
+      isByTable ? String.format("table (%s)", tableName) : "cluster",
+      total, sumMultiplier, minCostNeedBalance, maxRunningTime / 1000);
+    if (LOG.isTraceEnabled()) {
+      LOG.trace("Balance decision detailed function costs={}", functionCost());
     }
```

Comment:
if the general status message of "I'm doing some balancing (or not)" is going to info level, should this detail of individual cost functions go to debug?

Reply:
That makes sense. yes.